pkg/metrics/instance/configstore/kv/consul/mock.go

```diff
@@ -181,8 +181,7 @@ func (m *mockKV) List(prefix string, q *consul.QueryOptions) (consul.KVPairs, *c
 
 	result := consul.KVPairs{}
 	for _, kvp := range m.kvps {
-		if strings.HasPrefix(kvp.Key, prefix) && kvp.ModifyIndex >= q.WaitIndex {
-			// unfortunately real consul doesn't do index check and returns everything with given prefix.
```

Comment:
This is another behavior of Consul that we're relying on also being true for the Mock client.